src/vm/gdbjit.cpp

```diff
@@ -1654,11 +1678,15 @@ void ClassTypeInfo::DumpDebugInfo(char* ptr, int& offset)
 
 void ArrayTypeInfo::DumpDebugInfo(char* ptr, int& offset)
 {
-    if (m_type_offset != 0)
+    if (m_is_visited && m_base_ptr == ptr)
     {
         return;
     }
-    if (m_elem_type->m_type_offset == 0)
+
+    m_base_ptr = ptr;
+    m_is_visited = true;
+
+    if (!m_elem_type->m_is_visited || m_elem_type->m_base_ptr != ptr)
```

Comment:
The ArrayTypeInfo::DumpDebugInfo checks the `m_is_visited && m_base_ptr == ptr` at the beginning, but you make a similar check for m_elem_type before the call. Could you please revisit all the DumpDebugInfo methods to follow the pattern of having the check at their beginning and remove all the pre-call checks? That would make the code shorter, cleaner and easier to read.

Reply:
Done.